### ISSUE TITLE
Supporting Vue.js

### ISSUE THREAD

**Issue by wushan** [amcharts/amcharts3#110]
Is there any chance that releasing a amcharts component for vue.js ? Since Angular and React have one.

**Comment by Pauan** [comment]
@wushan We'll consider it. However, nobody on the AmCharts team uses Vue, so we'll have a hard time supporting it.

**Comment by craigcampbell** [comment]
Any update on this?

**Comment by martynasma** [comment]
No, not yet. There are no immediate plans for Vue.js support, I'm afraid.

**Comment by abelovic** [comment]
+1 for Vue support. 

Vue currently has 63k+ stars on github and is just about as popular as React and Angular (which you support)

**Comment by martynasma** [comment]
@abelovic I hear you. We are a small team, though, and can't possibly support everything. Vue is on our radar. Even if we don't have immediate plans or resources to implement the support, we might come back to it in the future.

I hope you find this reasonable.

**Comment by jgn-epp** [comment]
Still hoping this will happen one day.

**Comment by VSG24** [comment]
I want this.

**Comment by deemaxx** [comment]
+1

**Comment by ropinheiro** [comment]
+1

**Comment by mitrm** [comment]
+1

**Comment by mitrm** [comment]
What do you use in the alternative amcharts, for vuejs?

**Comment by jgn-epp** [comment]
I still use amcharts and use watchers to call update functions.

**Comment by LobsterMan** [comment]
+1

**Comment by NikolasVer** [comment]
+1

**Comment by martynasma** [comment]
We've just published a tutorial on using amCharts with Vue.js:

https://www.amcharts.com/kbase/using-amcharts-vue-js-webpack/

I know it's not exactly the same as native component, but still might help someone.

**Comment by bragma** [comment]
@martynasma thanks for the tutorial, but please don't put vue out of your sights. :)

**Comment by martynasma** [comment]
@bragma  Noted ;)

**Comment by peluprvi** [comment]
And please don't forget to place it in https://github.com/vuejs/awesome-vue#charts

**Comment by romakov** [comment]
When I try create a chart by the tutorial, I get some error like this:

> TypeError: r.a.makeChart is not a function

**Comment by ekamgit** [comment]
@martynasma I am also receiving `TypeError: r.a.makeChart is not a function`.

**Comment by ailon** [comment]
@romakov @ekamgit are you just following the tutorial or do you get the error in your own/some more elaborate setup? If the latter, would it be possible for you to share some code so we can investigate?

**Comment by romakov** [comment]
`<template>
  <div id="app">
    <img src="./assets/logo.png">
    <h1>{{ msg }}</h1>

    <div id="chartdiv" style="width: 100%; height: 400px;"></div>

    <h2>Essential Links</h2>
    <ul>
      <li><a href="https://vuejs.org" target="_blank">Core Docs</a></li>
      <li><a href="https://forum.vuejs.org" target="_blank">Forum</a></li>
      <li><a href="https://chat.vuejs.org" target="_blank">Community Chat</a></li>
      <li><a href="https://twitter.com/vuejs" target="_blank">Twitter</a></li>
    </ul>
    <h2>Ecosystem</h2>
    <ul>
      <li><a href="http://router.vuejs.org/" target="_blank">vue-router</a></li>
      <li><a href="http://vuex.vuejs.org/" target="_blank">vuex</a></li>
      <li><a href="http://vue-loader.vuejs.org/" target="_blank">vue-loader</a></li>
      <li><a href="https://github.com/vuejs/awesome-vue" target="_blank">awesome-vue</a></li>
    </ul>
  </div>
</template>

<script>
import AmCharts from 'amcharts3'
import AmSerial from 'amcharts3/amcharts/serial'

export default {
  name: 'app',
  data () {
    return {
      msg: 'Welcome to Your Vue.js App'
    }
  },
  created () {
    AmCharts.makeChart("chartdiv",
      {
        "type": "serial",
        "categoryField": "type",
        "chartCursor": {},
        "graphs": [
          {
            "type": "column",
            "title": "Pizza types",
            "valueField": "sold",
            "fillAlphas": 0.8
          }
        ],

        "dataProvider": [
          { "type": "Margherita", "sold": 120 },
          { "type": "Funghi", "sold": 82 },
          { "type": "Capricciosa", "sold": 78 },
          { "type": "Quattro Stagioni", "sold": 71 }
        ]
      }
    );
  }
}
</script>`
I make component by your guide, but AmCharts is empty object without a 'makeChart' method

**Comment by ailon** [comment]
@romakov Looks like amCharts package is missing in your project. Are you sure you've installed it from npm?

```npm install amcharts/amcharts3 --save```

**Comment by romakov** [comment]
@ailon Yes, I did it many times, but I still get the same error. Let's try do it step by step

**Comment by peluprvi** [comment]
Try to place it inside a `this.$nextTick(() => { /* AmCharts.makeChart here */ })`.
That is the majority difference between your setup and mine.
The other thing is that I imported AmChart in my `main.js`, but I don't think this is the case.

**Comment by quillhon** [comment]
@abelovic try to change
`import AmCharts from 'amcharts3'
import AmSerial from 'amcharts3/amcharts/serial'`
to
`import 'amcharts3'
import 'amcharts3/amcharts/serial'`

and then use
`window.AmCharts.makeChart(...`

@martynasma Do u really tried the tutorial? The amcharts3 library doesn't support import/module method. The js file only place AmCharts in global(window) scope.

**Comment by martynasma** [comment]
@quillhon No, I didn't. It was written, verified and reverified by my colleague @ailon. I'm going to leave it here, since I'm not competent to comment on the topic. If you elaborate what you are trying to do, maybe someone in this thread will be able to figure out why it's not working.

**Comment by cartbeforehorse** [comment]
1. Frustrating that the documentation on your site suggests that amCharts does come with Vue support.  At the very least you could update your documentation to inform the world of the true situation.
2. I understand that you're close to a v4 release of amCharts, so you're probably prioritising that at the moment.  Nevertheless, and ETA would be much appreciated.  I would "really" like to use amCharts on my Vue development projects, but without this kind of support may have to find an alternative.

**Comment by icangku** [comment]
`<template>
    <div id="chartdiv" style="width: 100%; height: 500px;"></div>
</template>`

`<script>
import AmCharts from 'amcharts3'
import AmSerial from 'amcharts3/amcharts/serial'
require('amcharts3/amcharts/gantt.js')
require('amcharts3/amcharts/themes/black.js')
require('amcharts3/amcharts/plugins/export/export.js')
require('amcharts3/amcharts/plugins/export/export.css')`

`export default {
    created(){
        window.AmCharts.makeChart("chartdiv",
        { demo code of gantt chart from amcharts https://www.amcharts.com/demos/gantt-chart-dates/ `

My code doesn't work very well, I need to refresh so many times afterwards it appears, what's wrong ?
